### PR TITLE
rm unnecessary judgement

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
rm unnecessary `!initContainerNames.Has(status.Name)`
if `status.Name != name` is false,  `!initContainerNames.Has(status.Name)` must be false

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
 NONE
```

### CODE REVIEW

**Comment by xiaoanyunfei** [comment]
ping @resouer @mtaufen

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 1]
```diff
@@ -635,7 +635,7 @@ func (m *kubeGenericRuntimeManager) pruneInitContainersBeforeStart(pod *v1.Pod,
 	for name := range initContainerNames {
 		count := 0
 		for _, status := range podStatus.ContainerStatuses {
-			if status.Name != name || !initContainerNames.Has(status.Name) ||
```
Comment (mtaufen) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go]: @yjhong looks like you wrote this originally. What was the reason for checking `initContainerNames.Has` here? (naively looking it it, we're already iterating the names and statuses in this nested loop, so I'm not sure why we'd iterate the names again)

**Pull request merged**